### ISSUE TITLE
W4 — conducting-wall electron current-closure implementation

### ISSUE THREAD

**Issue by HyungseonSong-plasma** [HyungseonSong-plasma/moose-test-repo#215]
## Work registration

**Parent controller:** #211  
**Depends on:** #212 W1 — PASS/CLOSED; #213 W2 — PASS/CLOSED; #214 W3 — PASS/CLOSED  
**Role:** W4 — implement the minimum accepted conducting-wall current closure  
**Complexity:** C3  
**Phase:** IMPLEMENT / VALIDATE  
**Status:** **READY — W3 selected grounded-conductor sheath-unresolved closure**  
**Prospective EVR budget:** 0/3

## Authorized production contract

#214 selected only the classical electron-repelling, sheath-unresolved grounded-conductor branch:

```text
phi_s  = plasma-side FV/sheath-edge potential
phi_w  = prescribed grounded conductor potential = 0 V
dphi_s = phi_s - phi_w >= 0

cbar_e = sqrt(8*e*T_e,eV/(pi*m_e))
S_s    = exp(-dphi_s/T_e,eV)

Gamma_e,primary,out
  = n_e,s*cbar_e/4 * S_s

Gamma_e,net,out
  = Gamma_e,primary,out - Gamma_e,SEE,in
```

Initial contract freezes reflection fraction `delta_r=0` and semi-Maxwellian replenishment `sigma_rp=1`.

The primary sheath-aware collection owner **replaces** the historical unsuppressed A7 primary electron wall-loss role on the W4 production path. It must not be summed with that loss and must not be composed with #213's forced signed bulk drift.

## Required ownership

```text
bulk electron drift at physical walls = EXCLUDED
bulk electron diffusion wall residual = natural zero-flux
primary electron collection           = one W4 sheath-aware wall owner
ion wall current                       = accepted A6/Stage-6 owners
SEE electron source                    = accepted A8 owner, gamma_O2p=gamma_Op=0.05
wall potential                         = grounded, prescribed 0 V
floating conductor                     = out of scope
dielectric sigma_s                     = out of scope
```

## Critical implementation gate — plasma-side potential

The sheath barrier must use the **plasma-side FV state**, not the Dirichlet boundary face value:

```text
CORRECT: dphi_s = phi_plasma_side - phi_wall
FORBIDDEN: dphi_s = phi_Dirichlet_face - phi_wall = 0
```

Before production runtime, the implementation must prove which MOOSE/FV API evaluates the element-side state on a boundary face.

If the generic `FVFunctorNeumannBC`/functor evaluation surface cannot make that ownership unambiguous, implement a narrow Physics FV boundary object rather than using an ambiguous parsed functor.

## Implementation preference

Use the narrowest production owner that can prove the W3 semantics:

1. audit reusable FV boundary APIs / existing Physics objects;
2. prefer a dedicated Physics FVBC if it is required to obtain element-side `phi_s` and `n_e,s` deterministically;
3. keep the sheath equation in exactly one object/functor owner;
4. retain separate diagnostics rather than putting diagnostic-only logic into the residual.

### P0 — analytic / construction

Must prove:

```text
dphi=0            -> S_s=1
dphi/T_e >> 1     -> S_s -> 0
wrong-sign exponent rejected
phi source is plasma-side, not grounded face value
bulk signed wall drift remains excluded
historical unsuppressed A7 primary owner absent/replaced
exactly one primary electron owner
```

### P1 — static/unit/ownership

Check:

```text
Gamma units = m^-2 s^-1
normalized n_e residual conversion is exact
outward-positive sign maps correctly to the normalized FV equation
SEE remains separate and inward
no floating/dielectric/RF ownership enters W4
```

### P2 — exact-head `physics-opt --check-input`

Required before runtime.

### P3 — bounded real-physics discriminator

At minimum compare:

```text
historical unsuppressed control
vs
new sheath-aware grounded-wall closure
```

and record:

```text
primary electron current
ion current
SEE current where enabled
net predicted charge current
predicted vs measured Delta Q_volume
potential extrema
n_e / solved-energy positivity
Gauss reconstruction
nonlinear chronology
```

A small potential is not an acceptance criterion by itself.

## Electron-energy claim boundary

W4's primary acceptance claim is particle/current closure. The Stage-6 solved-energy equation and accepted 4 eV SEE energy source must remain connected and positive.

The new particle law must not coexist with an obviously contradictory unsuppressed primary-energy loss. Any retargeting of the primary energy wall owner must be explicit and attributable to the same collected population. W4 does not by itself claim a complete absolute sheath-energy-transmission model.

## Acceptance

PASS only when the authorized grounded-conductor sheath-unresolved mechanism is exercised on the real provenance-controlled `physics-opt` path and:

```text
construction ownership PASS
sign/unit limiting controls PASS
current -> charge causality closes quantitatively
Gauss consistency PASS
no duplicate electron wall owner exists
accepted ion/SEE semantics remain intact
```

## Stop rules

- no unrestricted signed bulk drift as production wall physics;
- no A6 forced current cancellation;
- no coefficient tuning to obtain small potential;
- no electron-attracting/inverse/SCL branch invention;
- no floating-wall or dielectric sigma_s implementation;
- no closure from static/check-input evidence alone.

**Comment by HyungseonSong-plasma** [comment]
## W4 terminal evidence — ACCEPTED

**Exact head:** `c0325406878dd1f9583aaf446cbda702a82451f8`  
**Required workflow:** Issue-215 grounded electron sheath particle closure, run `34716147402` — **SUCCESS**  
**Evidence artifact:** `issue215-grounded-sheath-particle-evidence`, artifact `10305540377`, SHA-256 digest `54ea2f7523055c0a2053cb1864ac9b34f7b090be33b13b240ac0aeb9c2657765`

### Acceptance evidence

- immutable Physics provenance: PASS
- P0 analytic/construction/negative controls: PASS
- exact-head build + `physics-opt --check-input`: PASS
- real one-cell sheath smoke: PASS
  - `T_e = 3.82184 eV`, `exp(-10/T_e)=0.07305514413899142`
  - zero-drop expected/measured `n = 0.9967398406119159 / 0.99673984061192`
  - 10 V plasma-side-drop expected/measured `n = 0.9997611066531662 / 0.99976110665317`
  - measured plasma-side potential = `10 V`
- bounded reactor discriminator: all four control/sheath × baseline/refined-dt cases `MEASURED_COMPLETE`
- scientific gates G01–G06: all PASS
- artifact disposition: `W4_PARTICLE_CURRENT_PRECURSOR_PASS`; `scientific_hard_pass=true`

Representative equal-time (`t=2e-10 s`, `dt=1e-10 s`) causal comparison:

- historical unsuppressed control: `phi_max = 2088.6075 V`, primary normalized electron wall rate `438266.5659`
- grounded sheath closure: `phi_max = 20.55936 V`, primary normalized electron wall rate `4003.9009`
- sheath `n_e,min = 1.2911834e18 m^-3`
- particle inventory defect/loss = `2.90e-10`
- charge-current defect/current scale = `2.97e-10`
- Gauss defect/current scale = `2.46e-13`
- `Q_volume = 5.4297488278065e-08 C`, Gauss reconstruction `5.4297488278033e-08 C`

Refined-dt equal-time sensitivity (`dt=5e-11 s`):

- `phi_max` relative difference = `1.05694%`
- primary current-rate relative difference = `4.42999%`
- sheath trajectory remains complete/positive with charge-current and Gauss closure passing.

### Exact-head non-regression

At the same head:

- ordinary Physics CI — SUCCESS
- governed science validation P0–P3 — SUCCESS
- #193 governed bounded A8 finite-SEE runtime — SUCCESS
- #27 governed bounded final Stage-6 integration runtime — SUCCESS

This accepts the **grounded, sheath-unresolved particle/current implementation** authorized by #214. It does not broaden the claim to floating conductors, dielectric `sigma_s`, RF ownership, inverse/SCL branches, or a complete absolute sheath-energy-transmission model. W5 #216 remains the owner for the bounded integrated multi-step acceptance/handoff required by #211.
